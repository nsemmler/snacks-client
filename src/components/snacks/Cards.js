import React from 'react'
import {CardGroup} from 'reactstrap';
import Card from './Card'

const Cards = ({props}) => {
  return (<CardGroup>
    <div className="d-flex align-content-around flex-wrap">
      {
        props.map(card => {
          return (<Card key={card.id} card={card}/>)
        })
      }
    </div>
  </CardGroup>)
}

export default Cards
