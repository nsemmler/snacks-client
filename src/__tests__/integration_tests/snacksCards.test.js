import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import Cards from '../../components/snacks/Cards'

describe('Snacks Cards Component', () => {
  test('renders consistently', () => {
    const props = [
      {
        id: 1,
        name: "Pork Rinds",
        description: "Mauris lacinia sapien quis libero. Nam dui. Proin leo odio, porttitor id, consequat in, consequat ut, nulla. Sed accumsan felis.",
        price: 8,
        img: "https://az808821.vo.msecnd.net/content/images/thumbs/0000398_salt-pepper-pork-rinds-2-oz_560.jpeg"
      },
      {
        id: 2,
        name: "Soup - Campbells Beef Noodle",
        description: "Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Etiam vel augue. Vestibulum rutrum rutrum neque.",
        price: 26,
        img: "https://images-na.ssl-images-amazon.com/images/I/71MavWF1P9L._SY550_.jpg"
      }
    ]

    const card = renderer.create(
      <MemoryRouter>
        <Cards props={ props }/>
      </MemoryRouter>
    ).toJSON()
    expect(card).toMatchSnapshot()
  })
})
