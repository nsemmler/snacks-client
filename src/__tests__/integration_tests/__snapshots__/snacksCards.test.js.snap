// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snacks Cards Component renders consistently 1`] = `
<div
  className="card-group"
>
  <div
    className="d-flex align-content-around flex-wrap"
  >
    <div
      className="col-lg-4 col-md-2 col-sm-2 card"
      sm="6"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          Pork Rinds
        </h5>
        <h6
          className="card-subtitle"
        >
          $
          8
        </h6>
      </div>
      <img
        alt="Card image cap"
        className="card-img"
        id="card_img"
        src="https://az808821.vo.msecnd.net/content/images/thumbs/0000398_salt-pepper-pork-rinds-2-oz_560.jpeg"
        width="100%"
      />
      <div
        className="card-body"
      >
        <p
          className="card-text"
        >
          Mauris lacinia sapien quis libero. Nam dui. Proin leo odio, porttitor id, consequat in, consequat ut, nulla. Sed accumsan felis.
        </p>
        <a
          href="/snack/1"
          onClick={[Function]}
        >
          <button
            className="btn btn-secondary btn-lg btn-block"
            onClick={[Function]}
            type={undefined}
          >
            More Info
          </button>
        </a>
      </div>
    </div>
    <div
      className="col-lg-4 col-md-2 col-sm-2 card"
      sm="6"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          Soup - Campbells Beef Noodle
        </h5>
        <h6
          className="card-subtitle"
        >
          $
          26
        </h6>
      </div>
      <img
        alt="Card image cap"
        className="card-img"
        id="card_img"
        src="https://images-na.ssl-images-amazon.com/images/I/71MavWF1P9L._SY550_.jpg"
        width="100%"
      />
      <div
        className="card-body"
      >
        <p
          className="card-text"
        >
          Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Etiam vel augue. Vestibulum rutrum rutrum neque.
        </p>
        <a
          href="/snack/2"
          onClick={[Function]}
        >
          <button
            className="btn btn-secondary btn-lg btn-block"
            onClick={[Function]}
            type={undefined}
          >
            More Info
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
